telemetry/admin/src/main/scala/io/buoyant/telemetry/admin/AdminTelemeter.scala

```diff
@@ -0,0 +1,121 @@
+package io.buoyant.telemetry.admin
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter
+import com.twitter.finagle.http.{MediaType, Request, Response}
+import com.twitter.finagle.stats.NullStatsReceiver
+import com.twitter.finagle.tracing.NullTracer
+import com.twitter.util._
+import io.buoyant.admin.Admin
+import io.buoyant.telemetry.Metric.{Counter, Gauge, HistogramSummary, Stat}
+import io.buoyant.telemetry.{Metric, MetricsTree, Telemeter}
+import java.io.OutputStream
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * AdminTelemeter provides the /admin/metrics.json admin endpoint, backed by a MetricsTree.  It
+ * does not provide a StatsReciever or Tracer.  Histograms are snapshotted into summaries at a
+ * regular interval and the values served on /admin/metrics.json are taken from the most recent
+ * summary snapshot.  Counter and gauge values are served live.
+ */
+class AdminTelemeter(
+  metrics: MetricsTree,
+  snapshotInterval: Duration,
+  timer: Timer
+) extends Admin.Handler with Telemeter with Admin.WithHandlers {
+
+  val adminHandlers: Admin.Handlers = Seq(
+    "/admin/metrics.json" -> this
+  )
+
+  val stats = NullStatsReceiver
+  def tracer = NullTracer
+
+  private[this] val started = new AtomicBoolean(false)
+  // only run at most once
+  def run(): Closable with Awaitable[Unit] =
+    if (started.compareAndSet(false, true)) run0()
+    else Telemeter.nopRun
+
+  private[this] def run0() = {
+    val task = timer.schedule(snapshotInterval) {
+      summarySnapshots = snapshotHistograms(metrics)
+    }
+
+    new Closable with CloseAwaitably {
+      override def close(deadline: Time): Future[Unit] = closeAwaitably(task.close(deadline))
+    }
+  }
+
+  def apply(request: Request): Future[Response] = {
+    val pretty = request.getBooleanParam("pretty", false)
+    val response = request.response
+    response.mediaType = MediaType.Json
+    response.withOutputStream(writeJson(_, pretty))
+    Future.value(response)
+  }
+
+  private[this] var summarySnapshots = Map.empty[Stat, HistogramSummary]
+
+  private[this] val json = new JsonFactory()
+  private[this] def writeJson(out: OutputStream, pretty: Boolean = false): Unit = {
+    val jg = json.createGenerator(out)
+    if (pretty) jg.setPrettyPrinter(new DefaultPrettyPrinter())
+    jg.writeStartObject()
+    val flattened = if (pretty)
+      flattenMetricsTree(metrics).sortBy(_._1)
+    else
+      flattenMetricsTree(metrics)
```

Comment:
style nit: this is awkard to read for me (indentation doesn't make control flow obvious)

preferred:
```scala
val flattened =
  if (pretty) ...
  else ...
```
or
```scala
val flattened =
  if (pretty) {
    ...
  } else {
    ...
  }
```